test/CsWin32Generator.Tests/CsWin32GeneratorTestsBase.cs

```diff
@@ -161,14 +171,23 @@ protected async Task CompileGeneratedFilesWithSourceGenerators(string outputPath
             syntaxTrees.Add(syntaxTree);
         }
 
-        // Add the assembly attribute for DisableRuntimeMarshalling
-        string disableRuntimeMarshallingSource = @"
+        bool canDisableRuntimeMarshalling = this.tfm is null ||
+            this.tfm.StartsWith("net6.0", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
docs seem to indicate the attribute was added in .NET 7: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.disableruntimemarshallingattribute?view=net-9.0